### ISSUE TITLE
Number of iterations should be an int

### ISSUE THREAD

**Issue by Lysxia** [QuickChick/QuickChick#247]
Currently QuickChick's settings type `Arg` uses `nat` which makes it dangerous to use high numbers.

**Comment by liyishuai** [comment]
`nat` is extracted to OCaml `int`:
https://github.com/QuickChick/QuickChick/blob/3d96244177fc380cbd2deeb24ed95616db0ee8af/src/ExtractionQC.v.cppo#L12

**Comment by Lysxia** [comment]
Yes but it uses the terrible unary encoding. I'm also working on making QuickChick executable natively in Coq.

**Comment by Lysxia** [comment]
Using `int` rather than `nat` when the unboundedness is not essential would help us avoid accidental quadratic (or exponential) complexity like in #252
